src/main/java/edu/cornell/kfs/module/ar/batch/service/impl/CuVerifyBillingFrequencyServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ *
+ * Copyright 2005-2019 Kuali, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package edu.cornell.kfs.module.ar.batch.service.impl;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.kuali.kfs.coa.businessobject.AccountingPeriod;
+import org.kuali.kfs.coa.service.AccountingPeriodService;
+import org.kuali.kfs.integration.cg.ContractsAndGrantsBillingAward;
+import org.kuali.kfs.integration.cg.ContractsAndGrantsBillingAwardAccount;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.ar.batch.service.impl.VerifyBillingFrequencyServiceImpl;
+import org.kuali.kfs.module.ar.businessobject.BillingFrequency;
+import org.kuali.kfs.module.ar.businessobject.BillingPeriod;
+import org.kuali.kfs.sys.util.KfsDateUtils;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+
+import edu.cornell.kfs.module.ar.batch.service.CuVerifyBillingFrequencyService;
+
+import java.sql.Date;
+
+public class CuVerifyBillingFrequencyServiceImpl extends VerifyBillingFrequencyServiceImpl implements CuVerifyBillingFrequencyService {
+    
+    private static final Logger LOG = LogManager.getLogger(CuVerifyBillingFrequencyServiceImpl.class);
+
+    @Override
+    public boolean validateBillingFrequency(ContractsAndGrantsBillingAward award) {
+        LOG.info("validateBillingFrequency: Called with just Award. Sending Award lastBilledDate = " + award.getLastBilledDate() + " to private method.");
+        return validateBillingFrequency(award, award.getLastBilledDate());
+    }
+
+    @Override
+    public boolean validateBillingFrequency(ContractsAndGrantsBillingAward award, ContractsAndGrantsBillingAwardAccount awardAccount) {
+        LOG.info("validateBillingFrequency: Called with Award and Award Account. Sending awardAccount currentLastBilledDate = " + awardAccount.getCurrentLastBilledDate() + " to private method.");
+        return validateBillingFrequency(award, awardAccount.getCurrentLastBilledDate());
+    }
+
+    private boolean validateBillingFrequency(ContractsAndGrantsBillingAward award, Date lastBilledDate) {
+        LOG.info("validateBillingFrequency: For Award/Proposal# = " + award.getProposalNumber() + " with lastBilledDate = " + lastBilledDate);
```

Comment:
should there be a null check for lastBilledDate?  I remember getting caught by some NPE on CGB area dates on a previous user story/

Reply:
Thanks, sounds good